### DISCUSSION TITLE
Eventbridge schema registry - dotnet messaging

### DISCUSSION THREAD

**Discussion by robvanpamel** [aws/aws-dotnet-messaging#131] [General]
When we are sending messages with dotnet to eventbridge including with aws dotnet messaging framework, we see that the real message structure is very often forgotten and moved due to serialization to a data, or payload field. 

When you look at the schema registry at that moment, you'll notice that most of the message being sent and captured by the schema discovery will look the same.  They have properties like 
- id (string)
- source (string)
- specversion (string)
- time (string)
- type (string)
- data (string)

But what we really care about is that data fields which is a string and contains the serialized data based on the type. Is there a way to 'fix' this in such a way that the real objects (representing our dotnet classes and not the envelope) and fields are shown there as well? Maybe, complex json types can be used? 

It might be a too general question so feel free to ignore this one.

**Comment by normj** [comment]
@robvanpamel I'm not sure exactly what you mean. Do you mean having the event content be solely the serialization of the .NET object? Somewhere there needs to be the metadata of the message like the message type.

That being said and I'm probably misunderstanding your goal. The serialization in this library is customizable by injecting your implementation of [IEnvelopeSerializer](https://github.com/awslabs/aws-dotnet-messaging/blob/main/src/AWS.Messaging/Serialization/IEnvelopeSerializer.cs) and [IMessageSerializer](https://github.com/awslabs/aws-dotnet-messaging/blob/main/src/AWS.Messaging/Serialization/IMessageSerializer.cs) service interfaces.
